File: LeetCode/MergeStringsAlternately.cs
using System;
using System.Text;
class Program
{
    public string MergeAlternately(string word1, string word2)
    {
        var word1Array = word1.ToCharArray();
        var word2Array = word2.ToCharArray();
        int word1ArrayLength = word1Array.Length;
        int word2ArrayLength = word2Array.Length;
        StringBuilder sb = new StringBuilder();
        int i = 0, j = 0;
        while (word1ArrayLength > 0 && word2ArrayLength > 0)
        {
            sb.Append(word1[i++]);
            sb.Append(word2[j++]);
            word1ArrayLength--;
            word2ArrayLength--;
        }

        while (word1ArrayLength > 0)
        {
            sb.Append(word1[i++]);
            word1ArrayLength--;
        }

        while (word2ArrayLength > 0)
        {
            sb.Append(word2[j++]);
            word2ArrayLength--;
        }
        return sb.ToString();
    }
}

File: LeetCode/MergeStringsAlternately.cpp
#include <bits/stdc++.h>
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int word1Length = word1.length();
        int word2Length = word2.length();
        
        string res ="";
        int i=0,  j=0;
        while(word1Length >0 && word2Length >0){
            word1Length--; word2Length--;
            res+=word1[i++];
            res+=word2[j++];            
        }
        
        while(word1Length>0){
            word1Length--;
            res+=word1[i++];
        }
        
        while(word2Length>0){
            word2Length--;
            res+=word2[j++];
        }
        return res;
    }
};